src/main/java/com/getbase/sync/SyncProcess.java

```diff
@@ -40,6 +41,27 @@ public boolean run() {
         return true;
     }
 
+    /**
+     * @param timeToRunInMillis - maximum time to run for sync process
+     * @return true if sync was full - there are no more elements to fetch, false otherwise
+     */
+    public boolean run(final long timeToRunInMillis) {
```

Comment:
Let's keep only this version of method but take `timeToRunInMillis` from the field.